### ISSUE TITLE
[1849 variants] server caught in a refreshing loop

### ISSUE THREAD

**Issue by Adrianows** [tobymao/18xx#11138]
Game https://18xx.games/game/173565

Error message 
>The server did not accept this action due to: Illegal action: Blocking step Lay/Upgrade Track cannot process action 246... refreshing

action 246 is
  {
      "type": "lay_tile",
      "entity": "CTL",
      "entity_type": "corporation",
      "id": 246,
      "created_at": 1723662227,
      "hex": "E3",
      "tile": "8-1",
      "rotation": 4
    },

For what it's worth Ive seen this twice now, the first time resolved itself by repeatedly inputting my actions.

Edit: Running with the all the new variants (except for delay IFT)

**Comment by Adrianows** [comment]
Example screenshot 
![Screenshot_20240814-172727](https://github.com/user-attachments/assets/37616a56-1130-403a-a9b1-b4c1dcf3bc80)

**Comment by michaeljb** [comment]
@philcampeau this game was ended, can this issue be closed?

**Comment by philcampeau** [comment]
Well, I'm not sure that the update that broke the game actually fixed that issue. I suppose we could close it and wait to see if there are other similar reports.

**Comment by jck420** [comment]
The "blocking step" bug is happening constantly in this game too: https://18xx.games/game/174799

Definitely connected to the Bonds variant. Attempting various voodoo fixes of waiting 10 seconds or 5 seconds in between actions, refreshing the page etc. Sometimes they work sometimes they don't. Nearly unplayable though.

**Comment by philcampeau** [comment]
Yes. The issue is definitely related to the Bonds variant. The problem I'm having fixing it is that it doesn't happen in my local dev environment, it only happens on the live server. So I need one of the main devs to log into the server via SSH to pull the logs live while the bug is occurring. 

Right now it's just an issue of aligning our schedules to be able to do it.

**Comment by RedNick** [comment]
Not sure if this is the same issue, but I'm getting 
`The server did not accept this action due to: Illegal action: Auto actions do not match... refreshing.`
when trying any action (including sending messages like the above poster)
Using the bonds variant too https://18xx.games/game/177820

**Comment by f-prettyland** [comment]
I think the same issue is happening here: https://18xx.games/game/176619
Setup: 1849, bonds variant.
Problem: irregularly while trying to take any action or undo we get the following error:
`The server did not accept this action due to: Illegal action: Blocking step Lay/Upgrade Track cannot process action 138... refreshing`.
To replicate in this game, try undo'ing the previous action.

Cannot replicate in hotseat mode, only in live play.

**Comment by mkkohls** [comment]
I'm getting the same issue on https://18xx.games/game/184505

**Comment by jlunders** [comment]
Same issue
 
`The server did not accept this action due to: Blocking step Lay/Upgrade Track cannot process action 513... refreshing`
Which was happening pretty consistently for all players.

Game: 1849 (E Trains and Bonds variants on)
Id: 190036

**Comment by philcampeau** [comment]
Alright, reopening. 

That's unfortunate. We'll keep digging.

**Comment by rabbit210** [comment]
We experience the same problem in this game with bonds. 
https://18xx.games/game/195678

And I noticed that Bonds are working wrong. According to the rules: 
If the corporation has issued its bond, it must pay L. 50 to the bank each operating turn. This happens after its financial activities step. If the corporate treasury does not contain enough money to cover this payment, emergency money raising is performed.
Corporations never paid interest in our game.

**Comment by jeroenh** [comment]
And another one: https://18xx.games/game/200322

**Comment by philcampeau** [comment]
I didn't link it properly, but it looks like this has finally been fixed with the update in #11682.

Please don't hesitate to reply to this bug report if you still run into this issue.
